src/datasets/arrow_dataset.py

```diff
@@ -331,66 +345,110 @@ def to_tf_dataset(
             label_cols (:obj:`List[str]` or :obj:`str`, default ``None``): Dataset column(s) to load as
                 labels. Note that many models compute loss internally rather than letting Keras do it, in which case it is
                 not necessary to actually pass the labels here, as long as they're in the input `columns`.
-            dummy_labels (:obj:`bool`, default ``False``): If no `label_cols` are set, output an array of "dummy" labels
-                with each batch. This can avoid problems with `fit()` or `train_on_batch()` that expect labels to be
-                a Tensor or np.ndarray, but should (hopefully) not be necessary with our standard train_step().
+            drop_columns (:obj:`List[str]` or :obj:`str` or :obj:`bool`, optional): Dataset column(s) to drop before
+                passing to the `collate_fn`. By default, no columns are dropped.
             prefetch (:obj:`bool`, default ``True``): Whether to run the dataloader in a separate thread and maintain
                 a small buffer of batches for training. Improves performance by allowing data to be loaded in the
                 background while the model is training.
+            error_on_missing (:obj:`bool`, default ``True``): Whether to return an error if one of the requested columns
+                is missing. Should be set to `False` if columns or label_cols are autodetected from a model.
 
         Returns:
             :class:`tf.data.Dataset`
         """
-
+        # region Pre-checks and warnings
         if config.TF_AVAILABLE:
             import tensorflow as tf
         else:
             raise ImportError("Called a Tensorflow-specific function but Tensorflow is not installed.")
 
+        if isinstance(tf.distribute.get_strategy(), tf.distribute.TPUStrategy):
+            logger.warning(
+                "Note that to_tf_dataset() loads the data with a generator rather than a full tf.data "
+                "pipeline and is not compatible with remote TPU connections. If you encounter errors, please "
+                "try using a TPU VM or, if your data can fit in memory, loading it into memory as a dict of "
+                "Tensors instead of streaming with to_tf_dataset()."
+            )
+        # endregion
+
+        # region Argument and default value handling
+        if collate_fn is None:
+            # Set a very simple default collator that just stacks things together
+            def default_collate_fn(features):
+                first = features[0]
+                batch = {}
+                for k, v in first.items():
+                    if isinstance(v, np.ndarray):
+                        batch[k] = np.stack([f[k] for f in features])
+                    else:
+                        batch[k] = np.array([f[k] for f in features])
+                return batch
```

Comment:
Can be defined outside of this function maybe?

Reply:
Sure, I'll move this! Also, the one change in the notebook is that I removed the `dummy_labels` arg in the calls to `to_tf_dataset()`, the rest is just useless metadata, yes.

Reply:
Moved it to a new `tf_utils` file, protected by a check to see if TF is available.

Reply:
Renamed!